src/MarkerCluster.js

```diff
@@ -370,29 +370,20 @@ L.MarkerCluster = L.Marker.extend({
 		    zoom = this._zoom,
 		    i, c;
 
-		if (zoomLevelToStart > zoom) { //Still going down to required depth, just recurse to child clusters
-			for (i = childClusters.length - 1; i >= 0; i--) {
-				c = childClusters[i];
-				if (boundsToApplyTo.intersects(c._bounds)) {
-					c._recursively(boundsToApplyTo, zoomLevelToStart, zoomLevelToStop, runAtEveryLevel, runAtBottomLevel);
-				}
-			}
-		} else { //In required depth
-
+		if (zoomLevelToStart <= zoom) {
 			if (runAtEveryLevel) {
 				runAtEveryLevel(this);
 			}
-			if (runAtBottomLevel && this._zoom === zoomLevelToStop) {
+			if (runAtBottomLevel && zoom === zoomLevelToStop) {
 				runAtBottomLevel(this);
 			}
+		}
 
-			//TODO: This loop is almost the same as above
-			if (zoomLevelToStop > zoom) {
-				for (i = childClusters.length - 1; i >= 0; i--) {
-					c = childClusters[i];
-					if (boundsToApplyTo.intersects(c._bounds)) {
-						c._recursively(boundsToApplyTo, zoomLevelToStart, zoomLevelToStop, runAtEveryLevel, runAtBottomLevel);
-					}
+		if (zoom < zoomLevelToStart || zoom < zoomLevelToStop) {
```

Comment:
Hi,

Unfortunately this logical-looking assumption is not true there:
https://github.com/Leaflet/Leaflet.markercluster/blob/master/src/MarkerClusterGroup.js#L1154
`this._topClusterLevel._recursively(bounds, previousZoomLevel, 0, function (c) { /* … */ })`

Not sure how to re-factor that line in order to always have `zoomLevelToStart <= zoomLevelToStop`…